tests/class-wp-test-rest-post-type-controller-testcase.php

```diff
@@ -153,6 +153,48 @@ protected function check_post_data( $post, $data, $context ) {
 			sort( $data[ $taxonomy->rest_base ] );
 			$this->assertEquals( $terms, $data[ $taxonomy->rest_base ] );
 		}
+
+		// test links
+		if ( $links ) {
```

Comment:
@danielbachhuber this is due to the fact that items in collections don't have the same links in them as getting the single item, that's to say, getting a single item via `dispatch` will return a REST_Response that you call `get_links()` on, however that's a slightly different format to an collection item's `_links` property.